liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -194,15 +194,45 @@ public class DevMojo extends LooseAppSupport {
     /**
      * Containerfile used to build an image to then start a container with
      */
-    @Parameter(alias="containerfile", property = "containerfile")
+    private File containerfile;
     private File dockerfile;
 
+    @Parameter(alias="containerfile", property = "containerfile")
+    public void setContainerfile(File containerfile) {
+        this.containerfile = containerfile;
+        this.dockerfile = containerfile;
+    }
+
+    @Parameter(alias="dockerfile", property = "dockerfile")
+    public void setDockerfile(File dockerfile) {
+        if (dockerfile != null && this.containerfile != null) {
```

Comment:
No problem with the precedence, but am I correct that if both are set, you're only going to get the warning message if the setContainerFile happens to have been called first?  If we wanted to fix that, there should be some logic to issue the warning message in setContainerFile if the dockerfile field is set, I think.

Reply:
Maven evaluates all of the parameters alphabetically, so setContainerFile() should always get run first. Don't really like relying on that behavior to guarantee things. Think we're going to move all of this logic in to an init() method based on some discussion.